maintainers/scripts/haskell/regenerate-hackage-packages.sh

```diff
@@ -1,23 +1,28 @@
 #! /usr/bin/env nix-shell
 #! nix-shell -i bash -p coreutils haskellPackages.cabal2nix-unstable git nix -I nixpkgs=.
 
-# This script is used to regenerate nixpkgs' Haskell package set, using a tool
-# called hackage2nix. hackage2nix looks at the config files in
-# pkgs/development/haskell-modules/configuration-hackage2nix and generates
-# a Nix expression for package version specified there, using the Cabal files
-# from the Hackage database (available under all-cabal-hashes) and its
-# companion tool cabal2nix.
+# This script is used to regenerate nixpkgs' Haskell package set, using the
+# tool hackage2nix from the nixos/cabal2nix repo. hackage2nix looks at the
+# config files in pkgs/development/haskell-modules/configuration-hackage2nix
+# and generates a Nix expression for package version specified there, using the
+# Cabal files from the Hackage database (available under all-cabal-hashes) and
+# its companion tool cabal2nix.
 #
 # Related scripts are update-hackage.sh, for updating the snapshot of the
 # Hackage database used by hackage2nix, and update-cabal2nix-unstable.sh,
 # for updating the version of hackage2nix used to perform this task.
 
 set -euo pipefail
 
+# To prevent hackage2nix fails because of encoding.
+# See: https://github.com/NixOS/nixpkgs/pull/122023
+export LC_ALL=C.UTF-8
```

Comment:
For me this line causes an issue and fails generation. Without the line it is working.

```
./maintainers/scripts/haskell/regenerate-hackage-packages.sh
/home/andreas/dev/repository/nixpkgs/maintainers/scripts/haskell/regenerate-hackage-packages.sh: line 21: warning: setlocale: LC_ALL: cannot change locale (C.UTF-8): No such file or directory
Starting hackage2nix to regenerate pkgs/development/haskell-modules/hackage-packages.nix ...
hackage2nix: /home/andreas/dev/repository/nixpkgs/pkgs/development/haskell-modules/hackage-packages.nix: withFile: invalid argument (invalid character)
```

Reply:
I assume this is caused by the fact that nixos 22.11 changed the generation of locales. Actually not sure how exactly to fix this.

Reply:
Not sure, I'm not using NixOS, only the package manager (2.8.1). In my case the fix would be to remove the line, but I don't know if this is now true for everybody, only my system or it can be done conditionally.

Reply:
iirc the `C` locale is very similar to an English locale. Maybe `en_US.UTF-8` is good enough and more portable?
Just a thought; I'm not an expert on this.

Reply:
`export LC_ALL=en_US.UTF-8` would work in my case.

Reply:
Ah, sorry. Then it’s not nixos in your case. Anyway: The question is: Is there a reasonable encoding that we can expect to find on every system? I guess en_US.UTF-8 is the safest bet? At least better than what we have?

Reply:
`en_US.UTF-8` is relatively likely to not be present on NixOS systems nowadays. Plain `C` should be present on any POSIX-conformant system? `C.UTF-8` doesn't exist on macOS where `en_US.UTF-8` is always available though.